partiql-eval/src/test/kotlin/org/partiql/eval/CatalogTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.partiql.eval
+
+import org.junit.jupiter.api.Test
+import org.partiql.eval.compiler.PartiQLCompiler
+import org.partiql.parser.PartiQLParser
+import org.partiql.planner.PartiQLPlanner
+import org.partiql.spi.Context
+import org.partiql.spi.catalog.Catalog
+import org.partiql.spi.catalog.Session
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.PErrorListener
+import org.partiql.spi.function.AggOverload
+import org.partiql.spi.function.FnOverload
+import org.partiql.spi.function.Parameter
+import org.partiql.spi.types.PType
+import org.partiql.spi.value.Datum
+import org.partiql.spi.value.Field
+import kotlin.math.absoluteValue
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class CatalogTest {
+    companion object {
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val compiler = PartiQLCompiler.standard()
+    }
+
+    @Test
+    fun absent_scalar_extension_function() {
+        val query = "SELECT abs(x) as a, foo_bar_baz(x) as b, FOO_BAR_BAZ(x) as C FROM << {'x':1}, {'x':2} >>"
+        val catalog = object : Catalog {
+            override fun getName(): String = "default"
+        }
+        val result = testQuery(query, catalog)
+        assert(result is TestResult.Failure)
+        val errs = (result as TestResult.Failure).err
+        assertTrue(errs.errors.isNotEmpty())
+        assertEquals(PError.FUNCTION_NOT_FOUND, errs.errors[0].code())
+    }
+
+    @Test
+    fun scalar_extension_function() {
+        val query = "SELECT abs(x) as a, foo_bar_baz(x) as b, FOO_BAR_BAZ(x) as C FROM << {'x':1}, {'x':2} >>"
+        val catalog = object : Catalog {
+            override fun getName(): String = "default"
+            override fun getFunctions(session: Session, name: String): Collection<FnOverload> {
+                return listOf(
+                    // Just a `abs(F64, F64) -> F64` with a different name
```

Comment:
nit: fn signature should be a single argument?

```suggestion
                    // Just a `abs(F64) -> F64` with a different name
```